collator/src/collator/execution_manager.rs

```diff
@@ -125,6 +125,9 @@ impl MessagesExecutor {
         let config = self.config.clone();
         let params = self.params.clone();
 
+        // collect touched account ids for accounts with executed transactions only
+        let mut touched_account_ids: Vec<HashBytes> = Vec::new();
```

Comment:
Can we just track unique accounts here via `FastHashSet`? There is no need to store all duplicates when account spams messages to itself